core/src/test/scala/unit/kafka/tools/StorageToolTest.scala

```diff
@@ -192,6 +192,66 @@ Found problem:
     } finally Utils.delete(tempDir)
   }
 
+  private def runFormatCommand(stream: ByteArrayOutputStream, directories: Seq[String]): Int = {
+    val metaProperties = new MetaProperties.Builder().
+      setVersion(MetaPropertiesVersion.V1).
+      setClusterId("XcZZOzUqS4yHOjhMQB6JLQ").
+      setNodeId(2).
+      build()
+    val bootstrapMetadata = StorageTool.buildBootstrapMetadata(MetadataVersion.latestTesting(), None, "test format command")
+    StorageTool.formatCommand(new PrintStream(stream), directories, metaProperties, bootstrapMetadata, MetadataVersion.latestTesting(), ignoreFormatted = false)
+  }
+
+  @Test
+  def testFormatSucceedsIfAllDirectoriesAreAvailable(): Unit = {
+    val availableDir1 = TestUtils.tempDir()
+    val availableDir2 = TestUtils.tempDir()
+    try {
+      val stream = new ByteArrayOutputStream()
+      assertEquals(0, runFormatCommand(stream, Seq(availableDir1.toString, availableDir2.toString)))
+      assertTrue(stream.toString().contains("Formatting %s".format(availableDir1)))
+      assertTrue(stream.toString().contains("Formatting %s".format(availableDir2)))
+    } finally {
+      Utils.delete(availableDir1)
+      Utils.delete(availableDir2)
+    }
+  }
+
+  @Test
+  def testFormatSucceedsIfAtLeastOneDirectoryIsAvailable(): Unit = {
+    val availableDir1 = TestUtils.tempDir()
+    val unavailableDir1 = TestUtils.tempFile()
+    try {
+      val stream = new ByteArrayOutputStream()
+      assertEquals(0, runFormatCommand(stream, Seq(availableDir1.toString, unavailableDir1.toString)))
+      assertTrue(stream.toString().contains("I/O error trying to read log directory %s. Ignoring...".format(unavailableDir1)))
+      assertTrue(stream.toString().contains("Formatting %s".format(availableDir1)))
+      assertFalse(stream.toString().contains("Formatting %s".format(unavailableDir1)))
+    } finally {
+      Utils.delete(availableDir1)
```

Comment:
the file created by `TestUtils.tempFile()` will be delete on jvm exit (https://github.com/apache/kafka/blob/trunk/clients/src/test/java/org/apache/kafka/test/TestUtils.java#L198), so we don't need to call delete here

Reply:
Good spot! I'll remove these.